src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -1682,11 +1694,22 @@ void QueryAnalyzer::evaluateScalarSubqueryIfNeeded(QueryTreeNodePtr & node, size
     auto * query_node = node->as<QueryNode>();
     auto * union_node = node->as<UnionNode>();
     if (!query_node && !union_node)
-        throw Exception(ErrorCodes::LOGICAL_ERROR,
+        throw Exception(
```

Comment:
This does not look valid or better readable than before. Consider to revert this line.

Reply:
Unintentional change, reverted it